cardano-cli/src/Cardano/CLI/EraBased/Run/CreateTestnetData.hs

```diff
@@ -0,0 +1,626 @@
+import qualified Data.Text as Text
+import           Data.Time (NominalDiffTime, UTCTime, addUTCTime, getCurrentTime)
+import           Data.Word (Word64)
+import           GHC.Generics (Generic)
+import           Lens.Micro ((^.))
+import           System.Directory (createDirectoryIfMissing)
+import           System.FilePath ((</>))
+import qualified System.Random as Random
+import           System.Random (StdGen)
+
+import           Crypto.Random (getRandomBytes)
+
+runGenesisKeyGenUTxOCmd
+  :: GenesisKeyGenUTxOCmdArgs
+  -> ExceptT GenesisCmdError IO ()
+runGenesisKeyGenUTxOCmd
+    Cmd.GenesisKeyGenUTxOCmdArgs
+    { Cmd.verificationKeyPath
+    , Cmd.signingKeyPath
+    } = do
+    skey <- liftIO $ generateSigningKey AsGenesisUTxOKey
+    let vkey = getVerificationKey skey
+    firstExceptT GenesisCmdGenesisFileError
+      . newExceptT
+      $ writeLazyByteStringFile signingKeyPath
+      $ textEnvelopeToJSON (Just skeyDesc) skey
+    firstExceptT GenesisCmdGenesisFileError
+      . newExceptT
+      $ writeLazyByteStringFile verificationKeyPath
+      $ textEnvelopeToJSON (Just vkeyDesc) vkey
+  where
+    skeyDesc, vkeyDesc :: TextEnvelopeDescr
+    skeyDesc = "Genesis Initial UTxO Signing Key"
+    vkeyDesc = "Genesis Initial UTxO Verification Key"
+
+
+runGenesisKeyGenGenesisCmd
+  :: GenesisKeyGenGenesisCmdArgs
+  -> ExceptT GenesisCmdError IO ()
+runGenesisKeyGenGenesisCmd
+    Cmd.GenesisKeyGenGenesisCmdArgs
+    { Cmd.verificationKeyPath
+    , Cmd.signingKeyPath
+    } = do
+    skey <- liftIO $ generateSigningKey AsGenesisKey
+    let vkey = getVerificationKey skey
+    firstExceptT GenesisCmdGenesisFileError
+      . newExceptT
+      $ writeLazyByteStringFile signingKeyPath
+      $ textEnvelopeToJSON (Just skeyDesc) skey
+    firstExceptT GenesisCmdGenesisFileError
+      . newExceptT
+      $ writeLazyByteStringFile verificationKeyPath
+      $ textEnvelopeToJSON (Just Key.genesisVkeyDesc) vkey
+  where
+    skeyDesc :: TextEnvelopeDescr
+    skeyDesc = "Genesis Signing Key"
+
+
+runGenesisKeyGenDelegateCmd
+  :: GenesisKeyGenDelegateCmdArgs
+  -> ExceptT GenesisCmdError IO ()
+runGenesisKeyGenDelegateCmd
+    Cmd.GenesisKeyGenDelegateCmdArgs
+    { Cmd.verificationKeyPath
+    , Cmd.signingKeyPath
+    , Cmd.opCertCounterPath
+    } = do
+    skey <- liftIO $ generateSigningKey AsGenesisDelegateKey
+    let vkey = getVerificationKey skey
+    firstExceptT GenesisCmdGenesisFileError
+      . newExceptT
+      $ writeLazyByteStringFile signingKeyPath
+      $ textEnvelopeToJSON (Just skeyDesc) skey
+    firstExceptT GenesisCmdGenesisFileError
+      . newExceptT
+      $ writeLazyByteStringFile verificationKeyPath
+      $ textEnvelopeToJSON (Just Key.genesisVkeyDelegateDesc) vkey
+    firstExceptT GenesisCmdGenesisFileError
+      . newExceptT
+      $ writeLazyByteStringFile opCertCounterPath
+      $ textEnvelopeToJSON (Just certCtrDesc)
+      $ OperationalCertificateIssueCounter
```

Comment:
All those
```
 firstExceptT GenesisCmdGenesisFileError
      . newExceptT
```
can be joined together for better readability

Reply:
Done in https://github.com/input-output-hk/cardano-cli/pull/497 and will be ported in this PR by merging https://github.com/input-output-hk/cardano-cli/pull/497.